src/sanity_checker.rs

```diff
@@ -137,84 +138,80 @@ pub fn check_config_option_sanity(config_options: &Input) -> Result<(), String>
-                {
-                    true => trace!("{} UDP Port is numeric. Found: {}", name, port),
-                    false => {
-                        error!(
-                            "{} UDP Listen Port is not numeric or out of the range of 1-65353. Found: {}",
-                            name, port
-                        );
-                        is_input_sane = false;
-                    }
-                }
+    if let Some(ports) = option_ports {
+        if ports.is_empty() {
+            error!("{} An empty listen ports vec has been provided", name);
+            is_input_sane = false;
+        }
+        for port in ports {
+            if port.eq(&0) {
+                error!("{} Listen Port is invalid as it is zero", name);
+                is_input_sane = false
             }
         }
-        None => (),
     }
     is_input_sane
 }
 
-fn check_ports_are_valid_with_host(option_ports: &Option<Vec<String>>, name: &str) -> bool {
+fn check_ports_are_valid_with_host(socket_addresses: &Option<Vec<String>>, name: &str) -> bool {
     let mut is_input_sane = true;
-    match option_ports {
-        Some(ports) => {
-            if ports.is_empty() {
-                return false;
-            }
-
-            if ports[0].is_empty() {
-                return true;
-            }
-
-            for port in ports {
-                // split the host and port
 
-                let split_port: Vec<&str> = port.split(':').collect();
-
-                if split_port.len() != 2 {
-                    error!(
-                        "{} Port is not in the format host:port. Found: {}",
-                        name, port
-                    );
+    if let Some(sockets) = socket_addresses {
+        if sockets.is_empty() {
+            error!("{} has been provided, but there are no socket addresses", name);
+            is_input_sane = false;
+        }
+        for socket in sockets {
+            let parse_socket = SocketAddr::from_str(socket);
+            match parse_socket {
+                Err(parse_error) => {
+                    error!("{}: Failed to validate that {} is a properly formatted socket: {}", name, socket, parse_error);
                     is_input_sane = false;
-                    continue;
-                }
-
-                match split_port[1].chars().all(char::is_numeric)
-                    && split_port[1].parse().unwrap_or(0) > 0
-                    && split_port[1].parse().unwrap_or(65536) < 65535
-                {
-                    true => trace!("{} UDP Port is numeric. Found: {}", name, port),
-                    false => {
-                        error!(
-                            "{} UDP Listen Port is not numeric or out of the range of 1-65353. Found: {}",
-                            name, port
-                        );
+                },
+                Ok(parsed_socket) => {
+                    if parsed_socket.port().eq(&0) {
```

Comment:
So because `127.0.0.1:0` etc is a valid SocketAddr, has to be handled as a special snowflake here.